pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -447,6 +449,14 @@ func buildStatefulSet(mdb mdbv1.MongoDB) (appsv1.StatefulSet, error) {
 		},
 	}
 
+	ownerReference := []metav1.OwnerReference{
```

Comment:
[nit] as it's a slice, we should use the plural form for naming `ownerReferences`